### PR TITLE
Add all/sel mode for cli.

### PR DESCRIPTION
**Pull request by shinyypig**
Add the cli function that users can select the books to download.

### CODE REVIEW

**Comment by yihong0618** [comment]
Thanks, will take a look later.
@frostming is it PR affect the `GUI`?

File: kindle_download_helper/cli.py [hunk 1]
```diff
@@ -18,6 +18,80 @@
 
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
+# download selected books for cli
+def download_selected_books(kindle, options):
+    # get all books and get the default device
```
Comment (yihong0618) [inline kindle_download_helper/cli.py:23]: print some info here for using waiting

File: kindle_download_helper/cli.py [hunk 2]
```diff
@@ -18,6 +18,80 @@
...
+        downlist = []
+        flag = True
+        for idx in indices:
+            if idx.isnumeric() == False:
+                if ':' in idx:
+                    # if is not a number, and ":" in it, then it is a range
+                    # decode the range
+                    idx_begin, idx_end = [int(i) for i in idx.split(':')]
+                    # append the range to downlist
+                    [downlist.append(i) for i in range(idx_begin-1, idx_end)]
```
Comment (yihong0618) [inline kindle_download_helper/cli.py:55]: do not use list comprehension here maybe better

File: kindle_download_helper/cli.py [hunk 3]
```diff
@@ -18,6 +18,80 @@
...
+                else:
+                    # if is not a number, and no ":" in it, then it is an error
+                    print('Input error, please input numbers!!!')
+                    flag = False
+                    break
+            else:
+                # if is a number, then append it to downlist
+                downlist.append(int(idx)-1)
+        if flag == False:
```
Comment (yihong0618) [inline kindle_download_helper/cli.py:64]: if not flag:

**Review by yihong0618 — APPROVED** [review]
Thanks

File: kindle_download_helper/config.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 from kindle_download_helper.user_agents import USER_AGENTS
 
-DEFAULT_OUT_DIR = "DOWNLOADS"
+DEFAULT_OUT_DIR = "tmp"
```
Comment (yihong0618) [inline kindle_download_helper/config.py:5]: do not change dir here since we have other old users

**Review by yihong0618 — APPROVED** [review]

**Comment by yihong0618** [comment]
@shinyypig Thanks can you do some tiny change?

**Comment by shinyypig** [comment]
Sorry for this change，I thought this change is only valid for myself. I will fix this and improve the codes acoording to your suggestions tonight.在 2023年1月16日，20:21，yihong ***@***.***> 写道：﻿
@shinyypig Thanks can you do some tiny change?

**Comment by yihong0618** [comment]
Thank you very much.

**Pull request merged**